com.hazelcast.jet.sql.impl.connector.map.JoinByEquiJoinProcessorTest [HZ-1492]

_master_ (commit 2ad2bbbea3e55dfe04cfa97e884a2f4bfe59fe15)

Failed on MicrosoftJDK17:
- https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MicrosoftJDK17/30/testReport/com.hazelcast.jet.sql.impl.connector.map/JoinByEquiJoinProcessorTest/test_innerJoin/
- https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MicrosoftJDK17/30/testReport/com.hazelcast.jet.sql.impl.connector.map/JoinByEquiJoinProcessorTest/when_innerJoinProjectedByProjector_then_modified/

It fails in variants:
- `test_innerJoin`
- `when_innerJoinProjectedByProjector_then_modified`

<details><summary>Stacktrace (for test_innerJoin):</summary>

```
java.lang.AssertionError: 
processor output in mode "snapshots disabled, inboxLimit=1" doesn't match , expected:<expected:
[1, value]> but was:<actual:
>
	at com.hazelcast.jet.core.test.JetAssert.throwNotEqual(JetAssert.java:95)
	at com.hazelcast.jet.core.test.JetAssert.assertEquals(JetAssert.java:71)
	at com.hazelcast.jet.core.test.TestSupport.assertExpectedOutput(TestSupport.java:815)
	at com.hazelcast.jet.core.test.TestSupport.lambda$expectOutputs$4(TestSupport.java:345)
	at com.hazelcast.jet.core.test.TestSupport.runTest(TestSupport.java:807)
	at com.hazelcast.jet.core.test.TestSupport.run(TestSupport.java:632)
	at com.hazelcast.jet.core.test.TestSupport.assertOutput(TestSupport.java:423)
	at com.hazelcast.jet.core.test.TestSupport.expectOutputs(TestSupport.java:344)
	at com.hazelcast.jet.core.test.TestSupport.expectOutput(TestSupport.java:325)
	at com.hazelcast.jet.sql.impl.connector.map.JoinByEquiJoinProcessorTest.runTest(JoinByEquiJoinProcessorTest.java:168)
	at com.hazelcast.jet.sql.impl.connector.map.JoinByEquiJoinProcessorTest.test_innerJoin(JoinByEquiJoinProcessorTest.java:77)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details>

<details><summary>Standard output (for test_innerJoin):</summary>

```
Started Running Test: test_innerJoin
### Running the test, mode=snapshots disabled, inboxLimit=1
03:47:41.512579918 Input-0: [[1]]
03:47:41.513038510 Input processed, calling complete()
03:47:42,052  INFO |test_innerJoin| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
03:47:42,052  INFO |test_innerJoin| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
03:47:42,052  INFO |test_innerJoin| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
03:47:42,052  INFO |test_innerJoin| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/map]
BuildInfo right after test_innerJoin(com.hazelcast.jet.sql.impl.connector.map.JoinByEquiJoinProcessorTest): BuildInfo{version='5.2.0-SNAPSHOT', build='20220921', buildNumber=20220921, revision=2ad2bbb, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'test_innerJoin(com.hazelcast.jet.sql.impl.connector.map.JoinByEquiJoinProcessorTest):'
03:47:40, accumulated pauses: 1096 ms, max pause: 528 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/JoinByEquiJoinProcessorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/JoinByEquiJoinProcessorTest.java
@@ -22,13 +22,13 @@ import com.hazelcast.jet.core.test.TestSupport;
 import com.hazelcast.jet.sql.SqlTestSupport;
 import com.hazelcast.jet.sql.impl.JetJoinInfo;
 import com.hazelcast.jet.sql.impl.connector.keyvalue.KvRowProjector;
-import com.hazelcast.sql.impl.row.JetSqlRow;
 import com.hazelcast.map.IMap;
 import com.hazelcast.sql.impl.expression.ColumnExpression;
 import com.hazelcast.sql.impl.expression.ConstantExpression;
 import com.hazelcast.sql.impl.expression.Expression;
 import com.hazelcast.sql.impl.extract.GenericQueryTargetDescriptor;
 import com.hazelcast.sql.impl.extract.QueryPath;
+import com.hazelcast.sql.impl.row.JetSqlRow;
 import com.hazelcast.sql.impl.type.QueryDataType;
 import org.apache.calcite.rel.core.JoinRelType;
 import org.junit.Before;
@@ -63,7 +63,7 @@ public class JoinByEquiJoinProcessorTest extends SqlTestSupport {
 
     @BeforeClass
     public static void beforeClass() {
-        initialize(2, null);
+        initialize(1, null);
     }
 
     @Before

